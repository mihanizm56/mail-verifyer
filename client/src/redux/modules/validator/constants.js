export const VALIDATE_USER = "VALIDATE_USER";
export const ADD_VALIDATOR_USERNAME = "ADD_VALIDATOR_USERNAME";
export const ERROR_VALIDATE_FETCH = "ERROR_VALIDATE_FETCH";
