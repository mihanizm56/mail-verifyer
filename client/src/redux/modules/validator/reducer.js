import lensPath from "ramda/src/lensPath";
import set from "ramda/src/set";
import {
	ADD_VALIDATOR_USERNAME,
	ERROR_VALIDATE_FETCH,
	LOADING_START_VALIDATOR,
	LOADING_FINISH_VALIDATOR,
} from "./constants";

const initialState = {
	isLoading: false,
	error: null,
	username: "",
};

const loadingLens = lensPath(["isLoading"]);
const errorLens = lensPath(["error"]);
const usernameLens = lensPath(["username"]);

const validatorDataStorage = (state = initialState, action) => {
	switch (action.type) {
		case ADD_VALIDATOR_USERNAME:
			return set(usernameLens, action.payload, state);
		case LOADING_START_VALIDATOR:
			return set(loadingLens, true, state);
		case LOADING_FINISH_VALIDATOR:
			return set(loadingLens, false, state);
		case ERROR_VALIDATE_FETCH:
			return set(errorLens, action.payload, state);

		default:
			return state;
	}
};

export default validatorDataStorage;
