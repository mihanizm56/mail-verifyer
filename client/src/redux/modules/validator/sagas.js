import { call, put } from "redux-saga/effects";
import {
	addValidatorUsername,
	setValidateError,
	fetchLoadingValidatorStart,
	fetchLoadingValidatorFinish,
} from "./actions";
import { validateUserRequest } from "../../../services/api";

export function* validateUserEmailSaga(action) {
	try {
		const resultOfRequest = yield call(validateUserRequest, { token: action.payload });
		const { message, error, username } = resultOfRequest;

		if (error) {
			yield put(setValidateError(error));
		}

		if (message && !error && username) {
			yield put(addValidatorUsername(username));
		}
	} catch (error) {
		console.log("get error in validateUserEmailSaga", error);

		yield put(setValidateError(error));
	}
}
