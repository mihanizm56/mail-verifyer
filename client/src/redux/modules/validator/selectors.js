import { createSelector } from "reselect";

const errorState = state => state.validatorDataStorage.error;
const getUsername = state => state.validatorDataStorage.username;

export const getValidatorErrorState = createSelector(
	[errorState],
	state => state
);

export const getValidatorUsername = createSelector(
	[getUsername],
	username => username
);
