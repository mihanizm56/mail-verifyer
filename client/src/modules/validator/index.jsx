import React, { Component } from "react";
import { connect } from "react-redux";
import { withTranslation } from "react-i18next";
import { getValidatorErrorState, getValidatorUsername, fetchValidateUserToken } from "../../redux/modules/validator";
import { SuccessModal, ErrorModal, Loader } from "../../components";
import { REDIRECTION_URL_FOR_LINK_BUTTON } from "../../constants";

export class ValidatorContainer extends Component {
	componentDidMount = () => this.props.fetchValidateUserToken(this.props.token);

	render = () => {
		const { username, error, t: translate } = this.props;

		if (Boolean(error)) {
			return <ErrorModal errorText={error} />;
		}

		if (Boolean(username)) {
			return (
				<SuccessModal
					buttonClickRedirectsTo={REDIRECTION_URL_FOR_LINK_BUTTON}
					successText={translate("success.verified", { username })}
				/>
			);
		}

		return <Loader />;
	};
}

const mapStateToProps = store => ({
	error: getValidatorErrorState(store),
	username: getValidatorUsername(store),
});

const Wrapped = connect(
	mapStateToProps,
	{ fetchValidateUserToken }
)(ValidatorContainer);

export const Validator = withTranslation()(Wrapped);
