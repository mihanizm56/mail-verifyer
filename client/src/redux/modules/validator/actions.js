import { VALIDATE_USER } from "./constants";

export const fetchValidateUserEmail = payload => ({ type: VALIDATE_USER, payload });

export const setUsername = payload => ({ type: SET_USERNAME, payload });
