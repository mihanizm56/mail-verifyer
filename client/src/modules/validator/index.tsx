import React, { Component } from "react";
import { connect } from "react-redux";
import { compose } from "ramda";
import * as i18next from "i18next";
import { withTranslation } from "react-i18next";
import { getValidatorErrorState, getValidatorUsername, fetchValidateUserToken } from "../../redux/modules/validator";
import { SuccessModal, ErrorModal, Loader } from "../../components";
import { REDIRECTION_URL_FOR_LINK_BUTTON } from "../../constants";
// import { StoreType } from "../../redux/store";

interface IValidatorContainerProps {
	token?: string;
	username?: string;
	t: i18next.TFunction;
	error?: string;
	fetchValidateUserToken: (value: any) => void;
}

export class ValidatorContainer extends Component<any> {
	componentDidMount = () => this.props.fetchValidateUserToken(this.props.token);

	render = () => {
		const { username, error, t: translate } = this.props;

		if (Boolean(error)) {
			return <ErrorModal error={error} />;
		}

		if (Boolean(username)) {
			return (
				<SuccessModal
					buttonClickRedirectsTo={REDIRECTION_URL_FOR_LINK_BUTTON}
					successText={translate("success.verified", { username })}
				/>
			);
		}

		return <Loader />;
	};
}

const mapStateToProps = (store: any) => ({
	error: getValidatorErrorState(store),
	username: getValidatorUsername(store),
});

// const Wrapped = connect(
// 	mapStateToProps,
// 	{ fetchValidateUserToken }
// )(ValidatorContainer);

// export const Validator: any = withTranslation()(Wrapped);

export const Validator = compose(
	connect(
		mapStateToProps,
		{ fetchValidateUserToken }
	),
	withTranslation()
)(ValidatorContainer);
