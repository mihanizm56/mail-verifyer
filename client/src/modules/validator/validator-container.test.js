import React from "react";
import { shallow } from "enzyme";
import { ValidatorContainer } from "../validator";

describe("test ValidatorContainer", () => {
	beforeEach(() => {
		jest.clearAllMocks();
	});

	describe("test render contract", () => {
		it("snap test with all props", () => {
			const props = {
				fetchValidateUserToken: jest.fn(),
				token: "token",
				username: "test username",
				error: "",
				isLoading: false,
				t: value => value,
			};
			const component = shallow(<ValidatorContainer {...props} />);

			expect(component).toMatchSnapshot();
		});
		it("render ErrorModal when error prop was given", () => {
			const props = {
				fetchValidateUserToken: jest.fn(),
				token: "token",
				username: "test username",
				error: "test-error",
				isLoading: false,
				t: value => value,
			};
			const component = shallow(<ValidatorContainer {...props} />);
			const ErrorModal = component.find("ErrorModalMock");

			expect(ErrorModal.length).toEqual(1);
		});
		it("render SuccessModal when username prop was given", () => {
			const props = {
				fetchValidateUserToken: jest.fn(),
				token: "token",
				username: "test username",
				error: "",
				isLoading: false,
				t: value => value,
			};
			const component = shallow(<ValidatorContainer {...props} />);
			const SuccessModal = component.find("SuccessModalMock");

			expect(SuccessModal.length).toEqual(1);
		});
		it("render Loader when no username and no error was given", () => {
			const props = {
				fetchValidateUserToken: jest.fn(),
				token: "token",
				username: "",
				error: "",
				t: value => value,
			};
			const component = shallow(<ValidatorContainer {...props} />);
			const Loader = component.find("LoaderMock");

			expect(Loader.length).toEqual(1);
		});
	});
	describe("test props contract", () => {
		describe("test ErrorModal props", () => {
			it("set error prop", () => {
				const props = {
					fetchValidateUserToken: jest.fn(),
					token: "token",
					username: "",
					error: "test error",
					t: value => value,
				};
				const component = shallow(<ValidatorContainer {...props} />);
				const ErrorModal = component.find("ErrorModalMock");
				const errorProp = ErrorModal.prop("errorText");

				expect(errorProp).toEqual(props.error);
			});
		});
		describe("test SuccessModal props", () => {
			it("set buttonClickRedirectsTo prop", () => {
				const props = {
					fetchValidateUserToken: jest.fn(),
					token: "token",
					username: "test username",
					error: "",
					t: value => value,
				};
				const component = shallow(<ValidatorContainer {...props} />);
				const SuccessModal = component.find("SuccessModalMock");
				const buttonClickRedirectsToProp = SuccessModal.prop("buttonClickRedirectsTo");

				expect(buttonClickRedirectsToProp).toBeDefined();
			});
			it("set successText prop", () => {
				const props = {
					fetchValidateUserToken: jest.fn(),
					token: "token",
					username: "test username",
					error: "",
					t: value => value,
				};
				const component = shallow(<ValidatorContainer {...props} />);
				const SuccessModal = component.find("SuccessModalMock");
				const successTextProp = SuccessModal.prop("successText");

				expect(successTextProp).toBeDefined();
			});
		});
	});
	describe("test callback contract", () => {
		it("fetchValidateUserToken calls when component did mount", () => {
			const props = {
				fetchValidateUserToken: jest.fn(),
				token: "token",
				username: "test username",
				error: "",
				t: value => value,
			};
			const component = shallow(<ValidatorContainer {...props} />);

			expect(props.fetchValidateUserToken).toHaveBeenCalledWith(props.token);
		});
	});
});
