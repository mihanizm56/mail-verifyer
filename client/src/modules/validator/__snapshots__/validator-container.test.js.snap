// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`test ValidatorContainer test render contract snap test with all props 1`] = `
<SuccessModalMock
  buttonClickRedirectsTo="/send"
  successText="success.verified"
/>
`;
